multichain-testing/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -366,18 +310,47 @@ const advanceAndSettleScenario = test.macro({
       }
     };
 
-    await t.notThrowsAsync(() =>
-      retryUntilCondition(
+    let finalBlock;
+    await t.notThrowsAsync(async () => {
+      const q = await retryUntilCondition(
         () => queryClient.queryBalance(EUD, getUsdcDenom(eudChain)),
-        ({ balance }) =>
-          !!balance?.amount &&
-          BigInt(balance?.amount) > 0n &&
-          BigInt(balance.amount) < mintAmt,
+        ({ balance }) => {
+          if (!balance) return false; // retry
+          const value = BigInt(balance.amount);
+          if (value >= mintAmt) {
+            throw Error(`no fees were deducted: ${value} >= ${mintAmt}`);
+          }
+          if (value === 0n) return false; // retry
+          t.log('advance done', value, 'uusdc');
+          return true;
+        },
         `${EUD} advance available from fast-usdc`,
         // this resolves quickly, so _decrease_ the interval so the timing is more apparent
-        { retryIntervalMs: 500 },
-      ),
-    );
+        { retryIntervalMs: 500, maxRetries: 20 },
+      );
+      console.timeLog(`UX->${eudChain}`, 'rxd', q.balance?.amount);
+      ({ block: finalBlock } = await queryBlock(api));
+      console.timeLog(
+        `UX->${eudChain}`,
+        'final block',
+        finalBlock.header.height,
+        finalBlock.header.time,
+      );
+    });
+    console.timeEnd(`UX->${eudChain}`);
+    const blockDur =
+      Number(finalBlock.header.height) - Number(initialBlock.header.height);
+    const MAIN_BLOCK_SECS = 7;
+    const MAIN_MAX_DUR = 120; // product requirement: 2 min
+    const MARGIN_OF_ERROR = 0.2;
+    const mainWallClockEstimate = blockDur * MAIN_BLOCK_SECS;
+    t.log({
+      initialHeight: initialBlock.header.height,
+      finalHeight: finalBlock.header.height,
+      blockDur,
+      mainWallClockEstimate,
+    });
+    t.true(mainWallClockEstimate * (1 + MARGIN_OF_ERROR) <= MAIN_MAX_DUR);
```

Comment:
Can you walk me through this? Does it consider we're dealing with shorter block times here (`timeout_commit: 800ms`)?  I was expecting to see `finalBlock.header.time` used to make things equatable.